core/modules/order/store/actions.ts

```diff
@@ -19,6 +20,7 @@ const actions: ActionTree<OrderState, RootState> = {
    * @param {Order} order order data to be send
    */
   async placeOrder ({ commit, getters, dispatch }, order: Order) {
+    order.products = order.products.map(product => omit(product, ['configurable_options', 'configurable_children'])) as Order['products']
```

Comment:
Please just add the check `if (config.entities.optimize && config.entities.optimizeShoppingCart) {`